src/java/src/com/tigerbeetle/AccountsBatch.java

```diff
@@ -28,36 +27,40 @@ public AccountsBatch(Account[] accounts) {
 
     AccountsBatch(ByteBuffer buffer)
             throws RequestException {
+
         super(buffer);
 
         final var bufferLen = buffer.capacity();
 
         // Make sure the completion handler is giving us valid data
         if (bufferLen % Account.Struct.SIZE != 0)
-            throw new RequestException(RequestException.Status.INVALID_DATA_SIZE);
+            throw new AssertionError(
```

Comment:
Why make up a new AssertionError instead of using actual assertions?

https://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html#:~:text=An%20assertion%20is%20a%20statement,than%20the%20speed%20of%20light.

Reply:
Two reasons:

- It evaluates Expression1 and if it is false throws an AssertionError with **no detail message.**
- To add more information to the stack trace with an exception in our package `com.tigerbeetle.AssertionError`

I think we should inherit this exception from `java.lang.AssertionError` to make more sense for someone who's expecting an AssertionError.

Reply:
Let me check one thing... I didn't know that assertions could be disabled on the JVM, and we don't want this on ours assertions. I'm not sure if inheritance is going to be a good idea here.

Reply:
I had to do a quick research on this matter, to be honest, I knew very little about the `assert` keyword. 

To summarize it:

~It evaluates Expression1 and if it is false throws an AssertionError with no detail message.~
I was wrong about that, there is a syntax-sugar `assert x > 0 : "Error message!"`.

But anyway, it seems that `assert` has an undesirable semantics:

From [this post](https://www.yegor256.com/2016/06/17/dont-use-java-assertions.html):
"This is the first difference between exceptions and assertions. Exceptions will always be thrown, while assertions are not enabled by default. They are supposed to be turned on during testing and turned off in production. "

Inheritance would work fine here, we can do `throw new AssertionException()` even with assertions disabled, it is only the `assert` keyword that is turned off, not the AssertionException itself.